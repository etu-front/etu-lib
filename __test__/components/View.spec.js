import React from 'react'
import View from '../../src/components/View'
import { shallow, render } from 'enzyme'
import renderer from 'react-test-renderer'
import { decodeStream } from 'iconv-lite';

const setup = () => {
  const props = {
    className: 'customClass',
    onClick: jest.fn()
  }
  const wrapper = shallow(<View {...props}>hello</View>)
  return {
    props,
    wrapper
  }
}

describe('View', function() {
  const { props, wrapper } = setup()
  it('props', () => {
    const jsx = (
      <View
        row
        align={'flex-end'}
        flex={1}
        width={200}
        height={100}
        wrap
        justify={'stretch'}
        background={'red'}
        color={'white'}
        radius={5}
        style={{ lineHeight: '24px' }}
      >
        good
      </View>
    )
    const v = shallow(jsx)
    const style = v.prop('style')
    expect(v.text()).toEqual('good')
    expect(style).toHaveProperty('flexDirection', 'row')
    expect(style).toHaveProperty('alignItems', 'flex-end')
    expect(style).toHaveProperty('flexWrap', 'wrap')
    expect(style).toHaveProperty('justifyContent', 'stretch')
    expect(style).toHaveProperty('flex', 1)
    expect(style).toHaveProperty('width', 200)
    expect(style).toHaveProperty('height', 100)
    expect(style).toHaveProperty('background', 'red')
    expect(style).toHaveProperty('color', 'white')
    expect(style).toHaveProperty('borderRadius', 5)
    expect(style).toHaveProperty('lineHeight', '24px')
    const rendered = renderer.create(jsx)

    expect(rendered.toJSON()).toMatchSnapshot()

    const columnView = shallow(<View column>column</View>)
    expect(columnView.prop('style')).toHaveProperty('flexDirection', 'column')
    expect(View.displayName).toEqual('View')
    expect(View.Center.displayName).toEqual('View.Center')

    const layoutView = render(
      <View row>
        <View width={200}>1</View>
        <View flex={1} background={'#eee'} />
      </View>
    )
    expect(layoutView.find('div').length).toBe(2)
  })

  it('render layout', () => {
    const rendered = renderer.create(
      <View row>
        <View width={200}>1</View>
        <View flex={1} background={'#eee'} />
      </View>
    )
    expect(rendered.toJSON()).toMatchSnapshot()
  })

  it('render View.Center', () => {
    const tree = renderer.create(<View.Center>Hello</View.Center>).toJSON()
    expect(tree).toHaveStyleRule('align-items', 'center !important')
    expect(tree).toHaveStyleRule('justify-content', 'center !important')
    expect(tree).toMatchSnapshot()
  })

  it('should click event be called', () => {
    wrapper.simulate('click', { key: 'Click' })
    expect(props.onClick).toBeCalled()
  })

  it('should has custom classname', () => {
    expect(wrapper.is('.customClass')).toEqual(true)
  })

  it('layout 2', () => {
    const w = renderer.create(
      <View wrap row style={{ margin: '0 -10px'}}>
        <View height={100} flex={1} background='red' />
        <View height={100} flex={1} background='green' />
        <View height={100} flex={1} background='blue' />
      </View>
    ).toJSON()
    expect(w).toMatchSnapshot()
  })
})
