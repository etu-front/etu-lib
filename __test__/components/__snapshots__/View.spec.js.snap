// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View layout 2 1`] = `
.c0 {
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font: inherit;
  text-align: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  list-style: none;
  pointer-events: auto;
}

<div
  className="c0"
  style={
    Object {
      "alignItems": "flex-start",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "margin": "0 -10px",
    }
  }
>
  <div
    className="c0"
    style={
      Object {
        "background": "red",
        "flex": 1,
        "height": 100,
      }
    }
  />
  <div
    className="c0"
    style={
      Object {
        "background": "green",
        "flex": 1,
        "height": 100,
      }
    }
  />
  <div
    className="c0"
    style={
      Object {
        "background": "blue",
        "flex": 1,
        "height": 100,
      }
    }
  />
</div>
`;

exports[`View props 1`] = `
.c0 {
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font: inherit;
  text-align: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  list-style: none;
  pointer-events: auto;
}

<div
  className="c0"
  style={
    Object {
      "alignItems": "flex-end",
      "background": "red",
      "borderRadius": 5,
      "color": "white",
      "flex": 1,
      "flexDirection": "row",
      "flexWrap": "wrap",
      "height": 100,
      "justifyContent": "stretch",
      "lineHeight": "24px",
      "width": 200,
    }
  }
>
  good
</div>
`;

exports[`View render View.Center 1`] = `
.c1 {
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font: inherit;
  text-align: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  list-style: none;
  pointer-events: auto;
}

.c0 {
  -webkit-align-items: center !important;
  -webkit-box-align: center !important;
  -ms-flex-align: center !important;
  align-items: center !important;
  -webkit-box-pack: center !important;
  -webkit-justify-content: center !important;
  -ms-flex-pack: center !important;
  justify-content: center !important;
}

<div
  className="c0 c1"
  style={Object {}}
>
  Hello
</div>
`;

exports[`View render layout 1`] = `
.c0 {
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font: inherit;
  text-align: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  list-style: none;
  pointer-events: auto;
}

<div
  className="c0"
  style={
    Object {
      "alignItems": "flex-start",
      "flexDirection": "row",
    }
  }
>
  <div
    className="c0"
    style={
      Object {
        "width": 200,
      }
    }
  >
    1
  </div>
  <div
    className="c0"
    style={
      Object {
        "background": "#eee",
        "flex": 1,
      }
    }
  />
</div>
`;
